tests/karate/src/test/java/be/features/decentralizedCom/getMessagesById.feature

```diff
@@ -54,7 +55,40 @@ Feature: Request messages by id from other servers
     And eval heartbeat.params[lao.channel] = invalidMessageIds
 
     When mockServer.send(heartbeat)
-    And def getMessagesByIdMessages = mockServer.getMessagesByMethod('get_messages_by_id')
 
-    Then assert getMessagesByIdMessages.length == 0
+    Then assert mockServer.getGetMessagesById().length == 0
+
+
+  # Check that after the server confirms it received a message, sending a heartbeat containing that message id does not
+  # trigger a getMessagesById anymore
+  Scenario: Server should not request messages that it already has
+    Given def validRollCall = mockFrontend.createValidRollCall(lao)
+    And def validCreateRollCall =
+      """
+        {
+          "object": "roll_call",
+          "action": "create",
+          "id": '#(validRollCall.id)',
+          "name": '#(validRollCall.name)',
+          "creation": '#(validRollCall.creation)',
+          "proposed_start": '#(validRollCall.start)',
+          "proposed_end": '#(validRollCall.end)',
+          "location": '#(validRollCall.location)',
+          "description": '#(validRollCall.description)',
+        }
+      """
+
+    And mockFrontend.publish(validCreateRollCall, lao.channel)
+    And json answer = mockFrontend.getBackendResponse(validCreateRollCall)
+    And def message_id = mockFrontend.getPublishMessageId(validCreateRollCall)
+    And def messageIds = []
+    And eval messageIds.push(message_id)
```

Comment:
Is it possible to initialize in-line?
`And def messageIds = [message_id]`

Reply:
It seems like Karate has some problem parsing this as a JSON array, I get an error if I try to do it this way.